pydantic/_internal/_typing_extra.py

```diff
@@ -265,19 +265,23 @@ def eval_type_backport(
                 value, globalns, localns
             )
     except TypeError as e:
-        if not (isinstance(value, typing.ForwardRef) and is_backport_fixable_error(e)):
-            raise
-        try:
-            from eval_type_backport import eval_type_backport
-        except ImportError:
-            raise TypeError(
-                f'You have a type annotation {value.__forward_arg__!r} '
-                f'which makes use of newer typing features than are supported in your version of Python. '
-                f'To handle this error, you should either remove the use of new syntax '
-                f'or install the `eval_type_backport` package.'
-            ) from e
-
-        return eval_type_backport(value, globalns, localns, try_default=False)
+        if isinstance(value, typing.ForwardRef):
+            if is_backport_fixable_error(e):
+                try:
+                    from eval_type_backport import eval_type_backport
+                except ImportError:
+                    raise TypeError(
+                        f'You have a type annotation {value.__forward_arg__!r} '
+                        f'which makes use of newer typing features than are supported in your version of Python. '
+                        f'To handle this error, you should either remove the use of new syntax '
+                        f'or install the `eval_type_backport` package.'
+                    ) from e
+
+                return eval_type_backport(value, globalns, localns, try_default=False)
+
+            raise TypeError(f'Unable to evaluate type annotation {value.__forward_arg__!r}.') from e
```

Comment:
I'm trying to think of any `TypeError`s that could arise here:
- The ones related to new typing syntax is handled above (meaning https://github.com/pydantic/pydantic/issues/8307 can be closed)
- The original error reported in the issue (making use of generics on a non subscriptable class)
- Any others? If not, we could add a hint to this error message?:

```suggestion
            raise TypeError(f'Unable to evaluate type annotation {value.__forward_arg__!r}. Perhaps the type being used isn't subscriptable?') from e
```

Reply:
Nice, we might be able to say this fixes https://github.com/pydantic/pydantic/issues/8307 as well. Let's add this note for now - if someone reports a case where it doesn't make sense, we can further refine from there 👍